modules/ROOT/pages/secure-storage.adoc

```diff
@@ -0,0 +1,110 @@
+= Secure Storage Overview
+
+Anypoint Studio 7.10 and earlier versions stored the information of authenticated users in the workspace, meaning that you could have different authenticated users per workspace.
+
+Studio 7.11 introduces a Secure Storage facility provided by Eclipse to encrypt and save your credentials.
+
+Studio provides a master password provider that automatically generates a master password for all your user credentials. When adding a new credential, Studio uses your master password to encrypt and save that credential into a secure storage file in your file system.
+
+Because of the architecture of the secure storage, the authenticated users are now shared across all Eclipse-based workspaces and Anypoint Studio instances.
+
+== Master Password Providers
+
+The available master password providers depend on your operative system. Studio assigns a priority to the available master password providers and enables you to choose which one to use. By default, Studio assigns a higher level of priority to the password provider that uses your OS's underlying capabilities to generate and encrypt your master password.
+
+When adding a new credential to Studio, it automatically uses the master password generated by the password provider with the highest priority to encrypt your credential, and stores it in a secure storage file located in your filesystem.
+
+Windows Integration (64 bit)::
+Available on Windows. This provider uses a random value that is encrypted based on your Windows login information and stored in secure storage. This provider uses the master password automatically to decrypt your credentials.
+OS X Keystore Integration::
+Available on macOS. This provider uses a random value that is stored in the OS Keychain. You can check it using the Keychain Access utility. This provider uses the master password automatically to decrypt your credentials.
+Linux Integration (64 bit)::
+Available in Linux. This provider uses a random value and relies on the `libsecret` library that must be present and working on your Linux distribution. This provider uses the master password automatically to decrypt your credentials.
+UI Prompt::
+Available in Windows, macOS, and Linux. This provider prompts you to create a master password and uses it to encrypt your credentials. This provider requires you to enter your master password every time you open Studio.
+
+When adding a new credential for the first time, Studio notifies you that a new master password has been created. Additionally, Studio prompts you to create two password hints to use in case you forget your master password. The password hints are security questions that you must treat as a second password. If you lose your master password, your password hints are the only other method to recover your credentials.
```

Comment:
Yes, but I could use this to explain that by providing these hints a user may be able to recover an overwritten former master password, I'm basing this on Eclipse's help around this:

> Lost secure storage password or invalid secure storage password?
If you have specified password recovery questions/answers, you can use the command link General > Security > Secure Storage preference page to recover the master password. (...)  The recovered password is not displayed, but stored in the memory cache until the application is closed.

Based on the "invalid secure storage password" part, I assume this might be a good workaround for a user whose password was updated by another session. @mdaloia Is that correct?

Reply:
That's correct.

The questions are saved in the secure_storage file and the master password for a particular provider is encrypted with the combination of the answers of both questions. As the master password is saved encrypted inside the secure_storage file this is the only mechanism that allows to recover it, as in the case you described or, for example, if you make a backup of this file and want to use it after reinstalling your O.S.